ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -259,6 +265,25 @@ initNodeKernel args@NodeKernelArgs { registry, cfg, tracers
                                         ps_POLICY_PEER_SHARE_STICKY_TIME
                                         ps_POLICY_PEER_SHARE_MAX_PEERS
 
+    case genesisArgs of
+      GenesisDisabled    -> pure ()
+      GenesisEnabled ctx -> do
+        varLoEFragment <- newTVarIO $ AF.Empty AF.AnchorGenesis
+        setGetLoEFragment
+          (readTVar varGsmState)
+          (readTVar varLoEFragment)
+          ctx
+
+        void $ forkLinkedWatcher registry "NodeKernel.GDD" $
+          gddWatcher
+            cfg
+            (gddTracer tracers)
+            chainDB
+            (readTVar varGsmState)
+            -- TODO GDD should only consider (big) ledger peers
```

Comment:
This needs some discussion with the IOG networking team, for now, we can leave this as is.